src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidAccessTagCheck.java

```diff
@@ -0,0 +1,164 @@
+
+    private static final long serialVersionUID = 5197703822744690835L;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public InvalidAccessTagCheck(final Configuration configuration)
+    {
+        super(configuration);
+
+        final String highwayType = (String) this.configurationValue(configuration,
+                "minimum.highway.type", MINIMUM_HIGHWAY_TYPE_DEFAULT);
+        this.minimumHighwayType = Enum.valueOf(HighwayTag.class, highwayType.toUpperCase());
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check. Objects
+     * passed to this function have already been filtered by the tags.filter parameter in the
+     * configuration file.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return AccessTag.isNo(object) && ((object instanceof Edge) || (object instanceof Line))
+                && Edge.isMasterEdgeIdentifier(object.getIdentifier())
+                && !this.isFlagged(object.getOsmIdentifier()) && isMinimumHighway(object);
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        if (!isInMilitaryArea((LineItem) object))
+        {
+            this.markAsFlagged(object.getOsmIdentifier());
+            return Optional.of(this.createFlag(object,
+                    this.getLocalizedInstruction(0, object.getOsmIdentifier())));
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Checks if {@link LineItem} is inside an {@link Area} or {@link Relation} with tag
+     * {@code landuse=MILITARY} or tag key {@code military}.
+     *
+     * @param object
+     *            {@link LineItem} to check
+     * @return {@code true} if input {@link LineItem} is in an {@link Area} or {@link Relation} with
+     *         tag {@code landuse=MILITARY} or tag key {@code military}
+     */
+    private boolean isInMilitaryArea(final LineItem object)
+    {
+        for (final Area area : object.getAtlas()
+                .areas(area -> Validators.isOfType(area, LandUseTag.class, LandUseTag.MILITARY)
+                        || Validators.hasValuesFor(area, MilitaryTag.class)))
```

Comment:
This method is iterating over all the areas. Please use a method like `areasIntersecting` that would limit the search.

Reply:
The issue with using `areasIntersecting` is that it requires a polygon, and this check is working with `LineItem`s. As far as I am aware, `LineItem`s can be converted to polygons by either using their bounding box, or by connecting there first and last nodes. If this was done, false negatives could result if a road that would be flagged was near a military tagged area.    
`areasCovering` could be used, but would require iteration over all the `Location`s in the polyline of a `LineItem`. That is the best option I can come up with. I would love to hear if there is an even better alternative that you can think of.

Reply:
`LineItem` has a method called `bounds`, which will give you a `Polygon`. You can retrieve areas intersecting that and filter them down. e.g. `atlas.areasIntersecting(line.bounds(), area -> (line.intersects(areaPolygon) || areaPolygon.fullyGeometricallyEncloses(line.asPolyLine())) && (Validators.isOfType(area, LandUseTag.class, LandUseTag.MILITARY) || Validators.hasValuesFor(area, MilitaryTag.class)))`.

You should alsoe cache the result of `object.asPolyLine()`, because that method would be returning a new `PolyLine` object per call.